_posts/2016-03-16-AngularTS.md

```diff
@@ -0,0 +1,413 @@
+TypeScript offers an Object-oriented architecture experience, which means all code is defined in classes, interfaces and most of those classes can be instantiated into objects. It also supports encapsulation, which protects the data from unintended access and modification. 
+
+
+## Learning path of AngularTS
+If you’re no stranger to AngularJS you will notice that the structure remains the same. 
+Two way data binding, controllers, services, ... 
+But be aware that it has a different syntax in TypeScript. 
+I will show you the different best practices to implement these components. 
+
+
+## TypeScript Definition Files
+When using TS we have to reference to TSD files.
+These files describe the types defined in external libraries such as Angular. 
+To install the Angular TSD files we use typings.
+To use the typings manager we install it with:
+{% highlight text %}
+npm install typings --global
+{% endhighlight %}
+
+Afterwards install Angular with:
+{% highlight text %}
+typings install Angular --ambient --save
+{% endhighlight %}
+`--ambient --save` enables the flag and persists the selection in 'typings.json'
+
+All the installed TSD files are gathered in the typings folder.
+In the main.d.ts file you will see the references the application will use for Angular.
+Since Angular has multiple libraries, you can use the search command to find the proper definition you need.
+{% highlight text %}
+typings search Angular
+{% endhighlight %}
+It is possible that you have to declare the referencing on the top of your file.
+{% highlight text %}
+/// <reference path="../../typings/main.d.ts" />
+{% endhighlight %}
+
+## Modules
+
+### Angular Modules
+Modules are here to help us modularize our code.
+But it is important to know that if you’re not planning to share or declare related functionality.
+It is a best practice to use only one main module as the root of your application. 
+To let the module know of the existence of every component, they have to register themselves.
+Below every component declaration you will see a registration to the module. 
+When registering the module you have to add all the libraries you want to depend upon.
+In this example we inject the routing service for navigation. 
+
+{% highlight javascript %}
+module JWorks {
+    "use strict";
+
+    angular
+       .module("jworks360", ['ngRoute'])
+}
+{% endhighlight %}
+
+### Internal TypeScript Modules
+These modules are similar to namespaces.
+You can define an unique namespace around your code.
+This will encapsulate variables, interfaces and classes. 
+TypeScript supports sub namespaces for further encapsulation.
+
+{% highlight javascript %}
+module JWorks {
+    "use strict";
+}
+{% endhighlight %}
+
+#### Transpiled JavaScript
+
+{% highlight javascript %}
+var JWorks;
+(function(JWorks){
+
+})(JWorks || (JWorks = {}));
+{% endhighlight %}
+
+To encapsulate our code, the module will transpile to an IIFE(Immediately-Invoked Function Expression) around our components.
```

Comment:
Hate to say it, but maybe add a space between abbreviation and explanation? "IIFE (Ime ..."